src/Analyzers/CSharp/Analyzers/AddBraces/CSharpAddBracesDiagnosticAnalyzer.cs

```diff
@@ -265,23 +267,16 @@ private static IfStatementSyntax GetOutermostIfStatementOfSequence(SyntaxNode if
             IfStatementSyntax result;
             if (ifStatementOrElseClause.IsKind(SyntaxKind.ElseClause))
             {
-                result = (IfStatementSyntax)ifStatementOrElseClause.Parent;
+                result = (IfStatementSyntax)ifStatementOrElseClause.GetRequiredParent();
             }
             else
             {
                 Debug.Assert(ifStatementOrElseClause.IsKind(SyntaxKind.IfStatement));
                 result = (IfStatementSyntax)ifStatementOrElseClause;
             }
 
-            while (result != null)
-            {
-                if (!result.IsParentKind(SyntaxKind.ElseClause))
-                {
-                    break;
-                }
-
-                result = (IfStatementSyntax)result.Parent.Parent;
-            }
+            while (result.IsParentKind(SyntaxKind.ElseClause))
+                result = (IfStatementSyntax)result.GetRequiredParent().GetRequiredParent();
```

Comment:
Might just be me, but I much rather the previous form as far as readability goes - and I'm not talking about the braces 😛

I'm not even sure my brain can work out if these are equivalent.

Reply:
The problem with the prior form is that it was comparing something that could not be null to null. That informs the compiler the value is Nullable, which causes*more* warnings to deal with :-)